homeassistant/components/palazzetti/config_flow.py

```diff
@@ -48,3 +53,46 @@ async def async_step_user(
             data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: dhcp.DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle DHCP discovery."""
+
+        LOGGER.debug(
+            "DHCP discovery detected Palazzetti: %s", discovery_info.macaddress
+        )
+
+        self._async_abort_entries_match({CONF_HOST: discovery_info.ip})
+
+        await self.async_set_unique_id(dr.format_mac(discovery_info.macaddress))
+        self._abort_if_unique_id_configured()
+        self._discovered_device = PalazzettiClient(hostname=discovery_info.ip)
+        try:
+            await self._discovered_device.connect()
+        except CommunicationError:
+            LOGGER.exception("Communication error")
```

Comment:
Do we need to log in this case? I think that is not needed, as the discovery will just bail out.